packages/kbn-utils/src/schema/index.ts

```diff
@@ -433,7 +433,7 @@ export type ObjectSettingType<P extends Props> = Readonly<
   { [K in keyof P]: TypeOf<P[K]> }
 >;
 
-export class ObjectSetting<P extends Props> extends Setting<
+export class ObjectSetting<P extends Props = Props> extends Setting<
```

Comment:
This is just specifying a default type. It simplifies working with it when you expect a schema created using `schema.object()` as input, but don't care about the internals of the schema, e.g. this makes `foo` and `bar` equivalent:

```ts
function foo<T extends schema.ObjectSetting<schema.Props>>(o: T) {
}
function bar<T extends schema.ObjectSetting>(o: T) {
}

const objSchema = schema.object({ foo: schema.string() });
foo(objSchema);
bar(objSchema);
```